src/clif-util.rs

```diff
@@ -45,125 +41,179 @@ mod print_cfg;
-    // Parse command line arguments.
-    let args: Args = Docopt::new(USAGE)
-        .and_then(|d| {
-            d.help(true)
-                .version(Some(format!("Cranelift {}", VERSION)))
-                .deserialize()
-        })
-        .unwrap_or_else(|e| e.exit());
-
-    if args.flag_debug {
-        pretty_env_logger::init();
-    } else {
-        file_per_thread_logger::initialize(LOG_FILENAME_PREFIX);
-    }
+fn add_input_file_arg<'a>() -> clap::Arg<'a, 'a> {
+    Arg::with_name("file")
+        .required(true)
+        .multiple(true)
+        .value_name("file")
+        .help("Specify file(s) to be used for test")
+}
 
-    // Find the sub-command to execute.
-    let result = if args.cmd_test {
-        cranelift_filetests::run(args.flag_verbose, &args.arg_file).map(|_time| ())
-    } else if args.cmd_cat {
-        cat::run(&args.arg_file)
-    } else if args.cmd_filecheck {
-        rsfilecheck::run(&args.arg_file, args.flag_verbose)
-    } else if args.cmd_print_cfg {
-        print_cfg::run(&args.arg_file)
-    } else if args.cmd_compile {
-        compile::run(
-            args.arg_file,
-            args.flag_print,
-            &args.flag_set,
-            &args.flag_target,
-        )
-    } else if args.cmd_wasm {
-        #[cfg(feature = "wasm")]
-        let result = wasm::run(
-            args.arg_file,
-            args.flag_verbose,
-            args.flag_just_decode,
-            args.flag_check_translation,
-            args.flag_print,
-            &args.flag_set,
-            &args.flag_target,
-            args.flag_print_size,
-        );
-
-        #[cfg(not(feature = "wasm"))]
-        let result = Err("Error: clif-util was compiled without wasm support.".to_owned());
-
-        result
-    } else {
-        // Debugging / shouldn't happen with proper command line handling above.
-        Err(format!("Unhandled args: {:?}", args))
-    };
+fn add_verbose_flag<'a>() -> clap::Arg<'a, 'a> {
+    Arg::with_name("verbose").short("v").help("Be more verbose")
+}
 
-    if args.flag_time_passes {
-        print!("{}", timing::take_current());
+fn add_time_flag<'a>() -> clap::Arg<'a, 'a> {
+    Arg::with_name("time-passes")
+        .short("T")
+        .help("Print pass timing report for test")
+}
+
+fn add_set_flag<'a>() -> clap::Arg<'a, 'a> {
+    Arg::with_name("set")
+        .long("set")
+        .takes_value(true)
+        .multiple(true)
+        .help("Configure Cranelift settings")
+}
+
+fn add_target_flag<'a>() -> clap::Arg<'a, 'a> {
+    Arg::with_name("target")
+        .takes_value(true)
+        .long("target")
+        .help("Specify the Cranelift target")
+}
+
+fn add_print_flag<'a>() -> clap::Arg<'a, 'a> {
+    Arg::with_name("print")
+        .short("p")
+        .help("Print the resulting Cranelift IR")
+}
+
+/// Takes vector of clap values and converts the values to strings and puts values in the mut vector
+fn get_vec(mut_vec: &mut Vec<String>, argument_vec: Option<clap::Values<'_>>) {
```

Comment:
It appears all uses of `get_vec` are creating a new `Vec` to pass in for `mut_vec`. Consequently, it would simplify the code to make this function return a `Vec` (that it creates itself) rather than taking one as `&mut`.